src/coreclr/jit/codegenlinear.cpp

```diff
@@ -1848,7 +1848,17 @@ void CodeGen::genPutArgStkFieldList(GenTreePutArgStk* putArgStk, unsigned outArg
 
     // Evaluate each of the GT_FIELD_LIST items into their register
     // and store their register into the outgoing argument area.
-    const unsigned argOffset = putArgStk->getArgOffset();
+    const unsigned argOffset          = putArgStk->getArgOffset();
+    unsigned       preSpilledRegsSize = 0;
+#if FEATURE_FASTTAILCALL
+#ifdef TARGET_ARM
+    if (putArgStk->putInIncomingArgArea())
+    {
+        preSpilledRegsSize = genCountBits(regSet.rsMaskPreSpillRegs(true)) * REGSIZE_BYTES;
+    }
+#endif // TARGET_ARM
+#endif // FEATURE_FASTTAILCALL
+
```

Comment:
I think to avoid duplicating this code that makes up for the bias you can change this function to take `argOffset` as an argument and pass it from `genPutArgStk`. You will also need to change `genPutArgStk` in codegenxarch.cpp slightly.